### PR TITLE
Add AVHubert modules

### PR DESCRIPTION
**Pull request by yuta0306**

### CODE REVIEW

**Review by gunyarakun — APPROVED** [review]
LGTM!
Let Copilot review this.

File: pkg/avsr/src/avhubert/modeling_avhubert.py [hunk 1]
```diff
@@ -0,0 +1,726 @@
...
+                if e - span_start - keep_length - min_space > keep_length:
+                    new_parts.append((span_start + length + min_space, e))
+                return new_parts
+
+            parts = [(0, sz)]
+            min_length = min(lengths)
+            for length in sorted(lengths, reverse=True):
+                lens = np.fromiter(
+                    (e - s if e - s >= length + min_space else 0 for s, e in parts),
+                    np.int,
```
[the remark this answers is not in the document]
Comment (yuta0306) [reply pkg/avsr/src/avhubert/modeling_avhubert.py]: Partially applied

**Review by gunyarakun — APPROVED** [review]
LGTM, also!

**Review by lescidium — APPROVED** [review]
@yuta0306 @gunyarakun 
I was able to record a short video clip on my computer and pass it into this model and get an accurate transcription.
So we know it's working! LGTM.

It would be interesting to see how this model compares to just ReazonSpeech ASR model with no video. I am curious how much of an advantage is gained in transcription accuracy by having the visual.

My sample code to run the model:
```
from reazonspeech.avsr import AVHubertProcessor, AVHubertForConditionalGeneration

model_name = "/path/to/model"

# Load processor and model directly from Hugging Face
processor = AVHubertProcessor.from_pretrained(model_name)
model = AVHubertForConditionalGeneration.from_pretrained(model_name)

inputs = processor(raw_audio="test.webm", raw_video="test.webm", extract_mouth=True)
outputs = model.generate(**inputs, num_beams=5, max_new_tokens=256)
transcription = processor.decode(outputs[0], skip_special_tokens=True)

print(transcription)
```

The results from the terminal:
```
WARNING: All log messages before absl::InitializeLog() is called are written to STDERR
I0000 00:00:1759738736.629925  682597 gl_context_egl.cc:85] Successfully initialized EGL. Major : 1 Minor: 5
I0000 00:00:1759738736.632811  683576 gl_context.cc:369] GL version: 3.2 (OpenGL ES 3.2 Mesa 25.0.7-0ubuntu0.24.04.2), renderer: Mesa Intel(R) Graphics (RPL-P)
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
W0000 00:00:1759738736.636408  683558 inference_feedback_manager.cc:114] Feedback manager requires a model with a single signature inference. Disabling support for feedback tensors.
W0000 00:00:1759738736.643827  683563 inference_feedback_manager.cc:114] Feedback manager requires a model with a single signature inference. Disabling support for feedback tensors.
W0000 00:00:1759738736.645425  683563 landmark_projection_calculator.cc:186] Using NORM_RECT without IMAGE_DIMENSIONS is only supported for the square ROI. Provide IMAGE_DIMENSIONS or use PROJECTION_MATRIX.
きょうは10月6日です。
```

**Comment by yuta0306** [comment]
Thank you both for reviewing this PR.

I have some work to do before merging it:
- [ ] Update the root README
- [ ] bump the major version to v3.0.0

File: pkg/avsr/src/avhubert/decoder.py [hunk 2]
```diff
@@ -0,0 +1,1097 @@
...
+        if config.learned_pos:
+            self.pos_embed = LearnedPositionalEmbedding(
+                num_embeddings=config.max_position_embeddings,
+                embedding_dim=config.hidden_size,
+            )
+        else:
+            self.pos_embed = SinusoidalPositionalEmbedding(config=config)
+        self.layer_norm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.dropout = nn.Dropout(config.hidden_dropout)
+        self.layers = nn.ModuleList([AVHubertDecoderLayer(config) for _ in range(config.num_hidden_layers)])
```
Comment (gunyarakun) [inline pkg/avsr/src/avhubert/decoder.py]: `AVHubertDecoderStableLayerNorm` では `AVHubertDecoderLayer` に `layer_idx` を渡しているので、ここも同様に渡してあげるとよいかもしれません。

**Review by gunyarakun — APPROVED** [review]
LGTM!
